### PR TITLE
Hierarchical collections

### PR DESCRIPTION
**Pull request by tomkralidis**
This PR adds initial rudimentary support for hierarchical collections.  Note that this functionality may change in the future given the evolution of OGC API extension specifications for hierarchical collections.

From a user perspective, resources defined in configuration can be named with the hierarchy, i.e.:

```yaml
resources:
  naturalearth/lakes:
      title: 
```

which then is advertised in the API as:

- http://localhost:5000/collections/naturalearth/lakes
- http://localhost:5000/collections/naturalearth/lakes/items
- etc

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -228,6 +229,45 @@ Below is an example of how to integrate system environment variables in pygeoapi
...
+key of the resource name is the advertised collection identifier.  For example, given the following:
+
+.. code-block:: yaml
+
+  resources:
+    lakes:
+      ...
+
+
+The resulting collection will be made available at http://localhost:5000/collections/obs
```
Comment (GeoSander) [inline docs/source/configuration.rst:245]: Shouldn't that say `http://localhost:5000/collections/lakes`?

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -228,6 +229,45 @@ Below is an example of how to integrate system environment variables in pygeoapi
...
+All collections are published by default to http://localhost:5000/collections.  To enable
+hierarchical collections, provide the hierarchy in the resource key.  Given the following:
+
+.. code-block:: yaml
+
+  resources:
+    naturalearth/lakes:
+      ...
+
+The resulting collection will then be made available at http://localhost:5000/collections/naturalearth/obs
```
Comment (GeoSander) [inline docs/source/configuration.rst:256]: And this `http://localhost:5000/collections/naturalearth/lakes`?

File: docs/source/configuration.rst [hunk 3]
```diff
@@ -228,6 +229,45 @@ Below is an example of how to integrate system environment variables in pygeoapi
...
+
+.. note::
+
+  This functionality may change in the future given how hierarchical collection extension specifications
+  evolve at OGC.
+
+.. note::
+
+  Collection grouping is not available.  This means that while URLs such as http://localhost:5000/collections/naturalearth/obs
```
Comment (GeoSander) [inline docs/source/configuration.rst:265]: And also here obs = lakes?

**Review by GeoSander — APPROVED** [review]
Nice "low-hanging-fruit" kind of enhancement, @tomkralidis!
Apart for some typo in the docs (I think), it looks good and works for me. :+1:

**Comment by tomkralidis** [comment]
Thanks @GeoSander.  Fixes pushed to docs.

**Comment by tomkralidis** [comment]
Note related OAFeat discussion at https://gitter.im/opengeospatial/WFS_FES?at=62584163c4350025003cc25e

**Comment by tomkralidis** [comment]
Note hierarchical collections per this PR pass against CITE testing (ets-ogcapi-features10).

**Pull request merged**